backend/api_calls.py

```diff
@@ -728,6 +728,64 @@ def AddPlayerToChat(request, firebase):
   results.append(firebase.put('/games/%s/chatRooms' % game, chat, True))
 
 
+def Infect(request, firebase):
+  """Infect a player(otherPlayerId) and update access to chat rooms accordingly.
+
+  Validation:
+    playerId:
+    otherPlayerId:
+    gameId:
+
+    
+  """
+  results = []
+  valid_args = ['infectionId', 'playerId', 'otherPlayerId', 'gameId']
+  required_args = ['otherPlayerId', 'gameId']
+  ValidateInputs(request, firebase, required_args, valid_args)
+  playerId = request['playerId']
+  gameId = request['gameId']
+
+  if playerId and firebase.get('/games/%s/players/%s' % (gameId, playerId), None)['canInfect'] != "yes":
+    raise InvalidInputError('The alleged infector %s does not have the ability to infect.' % playerId)
+ 
+  infectionTime = int(time.time())
+  infecteePlayerId = request['otherPlayerId']
+  infected_data = {
+    'canInfect': 'yes',
+    'allegiance': 'horde',
+  }
+  results.append(firebase.put('/games/%s/players' % gameId, infecteePlayerId, infected_data))
+
+  game_chats = firebase.get('/games/%s/chatRooms' % gameId, None)
+  for chatId in game_chats.keys():
```

Comment:
The '.keys()' bit can be dropped. That's the default Python gives you when you iterate over a dict.